### PR TITLE
more Golang recommendations

### PR DESCRIPTION
**Pull request by nelztruss**
Pulled these suggestions out #38 that didn't get included because they would have inflated the initial PR.

### CODE REVIEW

File: web/server/go.md [hunk 1]
```diff
@@ -91,6 +91,12 @@ what response does the handler return?"
 This verifies the response/request responsibilities of the handler,
 and can be used to test edge cases with little test performance impact.
 
+#### Composing Handlers
+
+Handlers, and middlewares more generally, can greatly benefit from being written as ["adapters"](https://medium.com/@matryer/the-http-handler-wrapper-technique-in-golang-updated-bc7fbcffa702). You can then compose very complex flows out of many smaller parts, which is really just another expression of the [Unix philosophy](https://en.wikipedia.org/wiki/Unix_philosophy).
+
```
Comment (tinyels) [inline web/server/go.md]: I love the idea of creating adapters!
